### ISSUE TITLE
IftranRun producing multiple errors

### ISSUE THREAD

**Issue by afernandezody** [NCAR/ncl#159]
Hello,
My ncl installation is generating only a handful (28) of executables in the bin subdirectory. The log is about 40,000+ lines but the errors seem to originate from Iftran as they all look like:
```
/bin/sh: ../../../.././ncarg2d/src/libncarg/Iftran/Iftran: No such file or directory
make[5]: *** [Makefile:523: IftranRun] Error 127

/bin/sh: ../../../.././ncarg2d/src/libncarg/Iftran/Iftran: No such file or directory
make[5]: *** [Makefile:520: IftranRun] Error 127
```
and several others alike. [NOTE: Thus far I'm ignoring the errors of the kind 'Type mismatch' or 'Rank mismatch' because they don't seem to be the ones preventing the generation of the other binaries]. The problem probably originates from ncarg2d/src/libncarg but I'm not sure what to try. Thanks.
P.S. System is running CentOS8 and ncl is cloned from GitHub.
P.S. (2): I forgot to mention that the compilation is using GNU-11.1 installed at /opt rather than the defaults GNU-8.3.1 shipped with CentOS8.

**Comment by rbrownrigg** [comment]
Hi,

Do you have such a directory in your clone:  ......../ncarg2d/src/Iftran   ?

It sounds like for whatever reason, that executable is not getting built.
Perhaps check the log, or even cd directly into that directory and run
"make"

Rick

**Comment by afernandezody** [comment]
Hi @**rbrownrigg**, 
I think that you might be referring to .../ncarg2d/src/libncarg/Iftran.
When I invoke make, it's returning:
`make: *** No rule to make target '../../../.././common/src/libncarg_c/libncarg_c.a', needed by 'Iftran'.  Stop.
`
I'm unsure why it didn't build this library so let me try again.

**Comment by afernandezody** [comment]
Hi Rick,
It seems that the problems with Iftran are caused by the lack of the libncarg_c.a library. However, I don't understand why the library is not built. When I look at the log, the libncarg_c part reads
```
make[4]: Entering directory '/opt/praetorium/ncl_aarch64/common/src/libncarg_c'
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o bcopyswap.o bcopyswap.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o gsbytes.o gsbytes.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o logic32.o logic32.c
/opt/praetorium/G11/bin/gfortran -fPIC -fno-second-underscore -fno-range-check -fopenmp  -O   -c argget.f
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o options.o options.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o misc.o misc.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o ncarg_path.o ncarg_path.c
ncarg_path.c: In function â€˜_NGResolvePathâ€™:
ncarg_path.c:135:41: warning: implicit declaration of function â€˜endpwentâ€™ [-Wimplicit-function-declaration]
  135 |                                         endpwent();
      |                                         ^~~~~~~~
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o version.o version.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o error.o error.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o atoargv.o atoargv.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o usleep.o usleep.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o env.o env.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o trnspprt.o trnspprt.c
/opt/praetorium/G11/bin/gfortran -fPIC -fno-second-underscore -fno-range-check -fopenmp  -O    -c -o lblank.o lblank.f
/opt/praetorium/G11/bin/gfortran -fPIC -fno-second-underscore -fno-range-check -fopenmp  -O    -c -o iargct.o iargct.f
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o rwchinfl.o rwchinfl.c
/opt/praetorium/G11/bin/gcc -ansi -fPIC -fopenmp -std=c99  -O  -I../../.././include -I/usr/include -I/usr/include/freetype2  -DLinux  -DIBM -DNGVERSION='"6.6.2"'  -DNCLVERSION='"6.6.2"'  -DNCARGURL='"ngurl"'  -DSYSV -D_POSIX_SOURCE -D_XOPEN_SOURCE -DByteSwapped -DNeedFuncProto    -c -o sdbm.o sdbm.c
/opt/praetorium/G11/bin/gfortran -fPIC -fno-second-underscore -fno-range-check -fopenmp  -O    -c -o i1mach.o i1mach.f
/opt/praetorium/G11/bin/gfortran -fPIC -fno-second-underscore -fno-range-check -fopenmp  -O    -c -o r1mach.o r1mach.f
r1mach.f:167:27:

  167 |                CALL I1MCRA(SMALL, K, 16, 0, 0)
      |                           1
Error: Rank mismatch in argument â€˜aâ€™ at (1) (scalar and rank-1)
r1mach.f:168:27:

  168 |                CALL I1MCRA(LARGE, K, 32751, 16777215, 16777215)
      |                           1
Error: Rank mismatch in argument â€˜aâ€™ at (1) (scalar and rank-1)
r1mach.f:169:27:

  169 |                CALL I1MCRA(RIGHT, K, 15520, 0, 0)
      |                           1
Error: Rank mismatch in argument â€˜aâ€™ at (1) (scalar and rank-1)
r1mach.f:170:27:

  170 |                CALL I1MCRA(DIVER, K, 15536, 0, 0)
      |                           1
Error: Rank mismatch in argument â€˜aâ€™ at (1) (scalar and rank-1)
r1mach.f:171:27:

  171 |                CALL I1MCRA(LOG10, K, 16339, 4461392, 10451455)
      |                           1
Error: Rank mismatch in argument â€˜aâ€™ at (1) (scalar and rank-1)
make[4]: *** [<builtin>: r1mach.o] Error 1
make[4]: Leaving directory '/opt/praetorium/ncl_aarch64/common/src/libncarg_c'
Making ./common/src/bin
make[4]: Entering directory '/opt/praetorium/ncl_aarch64/common/src/bin'

```
so the problem seems to originate with r1mach.f, which I don't know what it does or why it's checking for so many architectures. Any suggestion would be welcomed. Thanks.

**Comment by rbrownrigg** [comment]
Things build out fine for me (just pulled from the repo HEAD and did a full
build). I looked at that module r1warp.f, and I don't see anything wrong
with those lines 167-169, etc.  The compiler is reporting garbage
characters. It looks like perhaps you are using the latest/greatest
(presumably!) gcc v11?  I'm working with v9.2.

Not sure what to advise...
Rick

**Comment by afernandezody** [comment]
Hi Rick,
The issue was that SMALL, LARGE, RIGHT, DIVER, and LOG10 are not defined for Cray T3E. Maybe it was just a typo, maybe someone meant these variable to be somewhat different. Whether that stops the compilation depends mainly on the compiler as you suggest (I still haven't seen any performance improvement with GNU-11 but it's way too early to draw any conclusion). I was able to make some progress as the compilation has now created 48 executables but ncl (and probably some others) is still missing. Need to keep troubleshooting.

**Comment by afernandezody** [comment]
At the end I was unable to compile it with GNU-11 due to its much stricter type checking for Fortran procedure arguments. Even trying -fallow-argument-mismatch turned out in a non-solution as it was producing segmentation faults all over the place.
